### PR TITLE
FOGL-8913: Readings count double on restart.

### PR DESCRIPTION
**Pull request by cyberwalk3r**

### CODE REVIEW

File: C/services/north/data_load.cpp [hunk 1]
```diff
@@ -652,6 +646,7 @@ void DataLoad::configChange(const string& category, const string& newConfig)
 		if (config.itemExists("source"))
 		{
 			setDataSource(config.getValue("source"));
+			m_lastFetched = 0; // Reset last fetch counter on source change
```
Comment (MarkRiddoch) [inline C/services/north/data_load.cpp:649]: Isn't this still going to reset the last fetched id when we have a configuration change? Surely we only wan to do this when the actual value of the source item changes?

Comment (cyberwalk3r) [reply C/services/north/data_load.cpp]: Yes you are correct, I have updated code to reset last fetched id only if source changes.

**Review by MarkRiddoch — APPROVED** [review]

**Comment by ashish-jabble** [comment]
@cyberwalk3r rebase with develop branch, so that CI checker will pass

**Comment by cyberwalk3r** [comment]
Closing this branch in favor of https://github.com/fledge-iot/fledge/pull/1430.

**Pull request closed without merging**